### PR TITLE
chore: updated release instructions

### PR DESCRIPTION
**Pull request by vvagaytsev**
**What this PR does / why we need it**:
This PR attempts to make the release guide a bit more clear.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -273,38 +273,38 @@ and does the following:
...
+2. Make the first pre-release:
+  * Reset `latest-release` to `master` with `git reset --hard origin/master`.
+  * Run `git log` to make sure that the latest commit is the expected one and there are no unwanted changes from master included in the release.
+  * Run `./scripts/release.ts preminor|prepatch`.
+  * Wait for the CI build job to get the binaries from the [Github Releases page](https://github.com/garden-io/garden/releases).
+3. Manual testing (using the pre-release/release binary)
+  * On **macOS** or **Linux**, run the `./scripts/test-release.sh <version>` script, where `<version>` should have the format `<major>.<minor>.<patch>-<preReleaseCounter>`, e.g. `0.12.38-0`. The script runs some simple tests to sanity check the release.
+  * On a **Windows** machine, run `garden deploy --dev vote --env remote` in the `vote` example project.
+    * If there are any issues with syncing, consider changing the `services[].devMode.sync[].mode` value(s) to `one-way-replica` and restart Garden.
```
Comment (thsig) [inline CONTRIBUTING.md:284]: "and restart Garden" -> "and restarting Garden"

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -273,38 +273,38 @@ and does the following:
...
+    * Change a file in the `vote` service and verify that the code synchronization was successful.
+    * Open the dashboard, verify that the initial page loads without errors.
+4. You might need to add some changes here. For example, if any other fix(es) should be included from master, or if there are any test failures. In that case ypou need a new pre-release:
```
Comment (thsig) [inline CONTRIBUTING.md:287]: "add some changes" -> "include some additional commits"

Comment (thsig) [inline CONTRIBUTING.md:287]: "add some changes" -> "include some additional commits"

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -273,38 +273,38 @@ and does the following:
...
+    * Checkout to the most recent pre-release branch, e.g. `1.2.3-0`, and cherry-pick the appropriate commits from `master`.
+    * Run `./scripts/release.ts prerelease` - it will generate new pre-release `1.2.3-1`.
```
Comment (thsig) [inline CONTRIBUTING.md:289]: "generate new" -> "generate a new"

Comment (thsig) [inline CONTRIBUTING.md:289]: "generate new" -> "generate a new"

**Review by thsig — CHANGES_REQUESTED** [review]
Looks good! Had a few text-related comments there.

File: CONTRIBUTING.md (hunk 1 above)
Comment (vvagaytsev) [reply CONTRIBUTING.md]: Fixed in https://github.com/garden-io/garden/pull/2986/commits/9862536356b8aac2ec8efa7376c3b77762e7dd76

File: CONTRIBUTING.md (hunk 2 above)
Comment (vvagaytsev) [reply CONTRIBUTING.md]: https://github.com/garden-io/garden/pull/2986/commits/9862536356b8aac2ec8efa7376c3b77762e7dd76

File: CONTRIBUTING.md (hunk 3 above)
Comment (vvagaytsev) [reply CONTRIBUTING.md]: https://github.com/garden-io/garden/pull/2986/commits/9862536356b8aac2ec8efa7376c3b77762e7dd76

Comment (vvagaytsev) [reply CONTRIBUTING.md]: Fixed in https://github.com/garden-io/garden/pull/2986/commits/9862536356b8aac2ec8efa7376c3b77762e7dd76

**Review by thsig — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (vvagaytsev) [reply CONTRIBUTING.md]: This has been fixed and force-pushed, the final changeset contains the correct version of the text.

**Pull request merged**